BrainPortal/cbrain_plugins/cbrain-plugins-base/userfiles/html_file/views/_html.html.erb

```diff
@@ -0,0 +1,52 @@
+
+<%
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2025
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+%>
+
+<% if page_is?('#show') %>
+  <%= link_to(
+      "Open in a separate tab",
+            url_for(
```

Comment:
I'm not sure why you are using such a low-level URL builder when we have higher level route helpers. Your URL should be built with:

```ruby
display_userfile_url(@userfile,
  :viewer         => :html,
  :content_viewer => "on",
  :file_name      => @userfile.name
)
```

This helper knows already about the action, the controller, and where to put the ID of the userfile inside it.